src/KeyringController.ts

```diff
@@ -412,28 +486,46 @@ class KeyringController extends EventEmitter {
    * Attempts to sign the provided message parameters.
    * Prefixes the hash before signing per the personal sign expectation.
    *
-   * @param {object} msgParams - The message parameters to sign.
-   * @param {object} opts - Additional signing options.
-   * @returns {Promise<Buffer>} The raw signature.
+   * @param msgParams - The message parameters to sign.
+   * @param opts - Additional signing options.
+   * @returns The raw signature.
    */
-  async signPersonalMessage(msgParams, opts = {}) {
+  async signPersonalMessage(
+    msgParams: MessageParams,
+    opts: Record<string, unknown> = {},
+  ): Promise<string> {
     const address = normalizeAddress(msgParams.from);
     const keyring = await this.getKeyringForAccount(address);
-    return await keyring.signPersonalMessage(address, msgParams.data, opts);
+    if (!keyring.signPersonalMessage) {
+      throw new Error(KeyringControllerError.UnsupportedSignPersonalMessage);
+    }
+
+    return await keyring.signPersonalMessage(
+      address,
+      msgParams.data as Hex,
+      opts,
+    );
   }
 
   /**
    * Get encryption public key
    *
    * Get encryption public key for using in encrypt/decrypt process.
    *
-   * @param {object} address - The address to get the encryption public key for.
-   * @param {object} opts - Additional encryption options.
-   * @returns {Promise<Buffer>} The public key.
+   * @param address - The address to get the encryption public key for.
+   * @param opts - Additional encryption options.
+   * @returns The public key.
    */
-  async getEncryptionPublicKey(address, opts = {}) {
+  async getEncryptionPublicKey(
+    address: string | Hex,
```

Comment:
Just an FYI, this will probably get collapsed to `string` if `Hex` is a subtype of `string`.

Reply:
[e36c134](https://github.com/MetaMask/KeyringController/pull/202/commits/e36c1348aadc882625deda681912d6b45786060e)